src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -190,13 +191,7 @@ public function addContent($content, $type = null)
     public function addHtmlContent($content, $charset = 'UTF-8')
     {
         // Use HTML5 parser if the content is HTML5 and the library is available
-        if (!$this->html5Parser
-            && class_exists(HTML5::class)
-            && '<!doctype html>' === strtolower(substr(ltrim($content), 0, 15))) {
-            $this->html5Parser = new HTML5(['disable_html_ns' => true]);
-        }
-
-        $dom = null !== $this->html5Parser ? $this->parseHtml5($content, $charset) : $this->parseXhtml($content, $charset);
+        $dom = null !== $this->html5Parser && strspn($content, " \t\r\n") === stripos($content, '<!doctype html>') ? $this->parseHtml5($content, $charset) : $this->parseXhtml($content, $charset);
```

Comment:
this is not left-trimming the content anymore to find the doctype. Is it expected ?

Reply:
yes it is, see strspn